src/main.rs

```diff
@@ -207,49 +250,93 @@ async fn get_new_bech32_address(client: &mut tonic_lnd::Client) -> Address {
+
+    let mut client = tonic_lnd::connect(
+        config.lnd_address,
+        &config.lnd_cert_path,
+        &config.lnd_macaroon_path,
+    )
+    .await
+    .expect("failed to connect");
+
+    println!("[DEBUG]: Connected to lnd");
+    let info = client
+        // All calls require at least empty parameter
+        .get_info(tonic_lnd::rpc::GetInfoRequest {})
         .await
-        .expect("failed to connect");
+        .expect("failed to get info ---- ");
 
+    // We only print it here, note that in real-life code you may want to call `.into_inner()` on
+    // the response to get the message.
+    println!("{:#?}", info);
     let mut handler = Handler::new(client).await?;
 
     if let Some(fee_rate) = args.next() {
-        let fee_rate = fee_rate.into_string().expect("fee rate is not UTF-8").parse::<u64>()?;
+        let fee_rate = fee_rate
+            .into_string()
+            .expect("fee rate is not UTF-8")
+            .parse::<u64>()?;
         let address = get_new_bech32_address(&mut handler.client).await;
 
+        println!("Address: {} ", address);
+
         let mut args = args.fuse();
         let mut scheduled_channels = Vec::with_capacity(args.size_hint().0 / 2);
         let mut wallet_amount = bitcoin::Amount::ZERO;
         while let Some(arg) = args.next() {
             match args.next() {
-                Some(channel_amount) => scheduled_channels.push(ScheduledChannel::from_args(arg, channel_amount)),
-                None => wallet_amount = bitcoin::Amount::from_str_in(arg.to_str().expect("wallet amount not UTF-8"), bitcoin::Denomination::Satoshi)?,
+                Some(channel_amount) => {
+                    scheduled_channels.push(ScheduledChannel::from_args(arg, channel_amount))
+                }
+                None => {
+                    wallet_amount = bitcoin::Amount::from_str_in(
+                        arg.to_str().expect("wallet amount not UTF-8"),
+                        bitcoin::Denomination::Satoshi,
+                    )?
+                }
             }
         }
 
+        println!("[DEBUG] Done scheduling payjoins");
+
         let scheduled_payjoin = ScheduledPayJoin {
             wallet_amount,
             channels: scheduled_channels,
             fee_rate,
         };
 
-        scheduled_payjoin.test_connections(&mut handler.client).await;
-
-        println!("bitcoin:{}?amount={}&pj=https://example.com/pj", address, scheduled_payjoin.total_amount().to_string_in(bitcoin::Denomination::Bitcoin));
-
-        handler.payjoins.insert(&address, scheduled_payjoin).expect("New Handler is supposed to be empty");
+        scheduled_payjoin
+            .test_connections(&mut handler.client)
+            .await;
+
+        println!(
+            "bitcoin:{}?amount={}&pj=https://example.com/pj",
+            address,
+            scheduled_payjoin
+                .total_amount()
+                .to_string_in(bitcoin::Denomination::Bitcoin)
+        );
```

Comment:
Amount string needs to go to a variable to not blow up the call. (Also ideally we would bump `bitcoin` version once it's realeased and use the newly added `display_in` method.)